### PR TITLE
Fix the Observe chart rendering and application mechanics to the cluster.

### PR DESCRIPTION
**Pull request by paroxp**
## What

A previous attempt at tidying the repo, caused by a misinterpretation of the intentions of this repo given its name, left the helm chart templating in a broken state.

This PR now rolls back those "tidying" changes to restore the intended helm chart templating functionality.

### CODE REVIEW

**Comment by issyl0** [comment]
😱

**Comment by issyl0** [comment]
Reviewing with `?w=1` in the URL makes it look a lot less scary.

**Comment by paroxp** [comment]
Yeah, sorry. I've moved the ordering when I was trying different things ;/ Happy to put it back so it looks even less scary :)

**Comment by chrisfarms** [comment]
I don't think we should be using a `helm fetch` here as it would make this repository different from the pattern of every user defining their deployments as a Chart, and require a unique pipeline.

I would expect this repo to define a Chart that references the upstream chart as a requirement so that it can use the same `chart -> template -> namespace` flow

File: requirements.lock [hunk 1]
```diff
@@ -0,0 +1,6 @@
+dependencies:
+- name: prometheus
+  repository: https://kubernetes-charts.storage.googleapis.com/
+  version: 7.4.1
```
Comment (paroxp) [inline requirements.lock]: Could/Should we upgrade? I liked how we got latest version out of concourse.

**Comment by blairboy362** [comment]
From the original description of the PR....


## What

Currently the configuration isn't being recognised by the `helm
template` as the indentation has been off.

It would appear that the values no longer need to be categorised... This
is really confusing and I didn't investigate further just yet... I have
managed to parse the templates correctly on my local machine.

This has been tested by running the following:

```sh
helm init --client-only

helm fetch stable/prometheus --untar --untardir chart

mkdir state

helm template \
  chart/prometheus/. \
  -f values.yaml.tmpl \
  --name alertmanager \
  --namespace observe \
  --output-dir state

cat state/prometheus/templates/alertmanager-configmap.yaml
```

This however leaves me with few concerns:

- The files are not being deleted by concourse, meaning deprecations
will not take affect on the cluster
- We should be pinning the version of everything (helm cli, chart,
kubernetes?!)
  - I'm not certain whether this is an API change or prometheus chart
change that's at fault
- We may need to think of running some sort of testing on concourse to
see if template parsing has happened as expected
  - Took a little while before I realised what's going on
- How on earth are we to separate config between charts if they happen
to have similar namespace? ie. `web.ingress.enabled`

Hopefully, this should unblock Observe for now...

./cc @chrisfarms

## How to review

- Run the above lines in your local environment

**Comment by blairboy362** [comment]
> The files are not being deleted by concourse, meaning deprecations
will not take affect on the cluster

Taken care of in https://github.com/alphagov/gsp-observe-alertmanager-spike/pull/1/commits/89c722cd8a068c560a5e03bb8416c0b314cf932c

> We should be pinning the version of everything

We are now.

> We may need to think of running some sort of testing on concourse

Yes, but we'll park that for now. That will be a bigger piece of work, I think.

> How on earth are we to separate config between charts if they happen
to have similar namespace?

This is no longer an issue now that we've reverted back to using charts as originally intended.

**Comment by paroxp** [comment]
It does feel less hacky and is less error prone. 👏

**Pull request merged**